### PR TITLE
Dashboard sections fetch

### PR DESCRIPTION
**Pull request by VincentCATILLON**
## Detailed purpose of the PR

This PR adds sections fetchable with scroll.

### Result and observation

You can fetch sections.

- [ ] **Breaking changes ?**
- [ ] **Extra dependency ?**

### Testing Strategy

- [ ] Already covered by tests
- [ ] Manual testing
- [x] Unit testing

### CODE REVIEW

**Comment by VincentCATILLON** [comment]
Waiting for @chrisdugne PR on mooc API

File: src/components/catalog.js [hunk 1]
```diff
@@ -64,24 +65,25 @@ class Catalog extends React.PureComponent<Props> {
...
 
   render() {
-    const {sections, onRefresh, isRefreshing = false} = this.props;
+    const {sections, onRefresh, isRefreshing = false, onScroll} = this.props;
 
     return (
       <FlatList
         refreshControl={<RefreshControl refreshing={isRefreshing} onRefresh={onRefresh} />}
-        data={sections.length > 0 ? sections : new Array(2).fill()}
+        data={sections.length > 0 ? sections : new Array(3).fill()}
```
Comment (Volubyl) [inline src/components/catalog.js]: on devrait pas mettre cette valeur en variable (env ou autre) ?

Comment (VincentCATILLON) [reply src/components/catalog.js]: done @Volubyl

**Comment by VincentCATILLON** [comment]
Do not merge until https://github.com/CoorpAcademy/coorpacademy/pull/14187 is merged

File: src/containers/catalog.js [hunk 2]
```diff
@@ -59,6 +70,40 @@ class Catalog extends React.PureComponent<Props, State> {
...
+
+  handleScroll = ({nativeEvent}: ScrollEvent) => {
+    const {layout, sections} = this.props;
+    const offsetY = nativeEvent.contentOffset.y;
+
+    if (offsetY !== this.offsetY && layout) {
+      this.offsetY = offsetY;
+      const offset = this.getOffset(offsetY);
+      const limit = this.getLimit();
+      const hasUnfetchedSections =
```
Comment (chrisdugne) [inline src/containers/catalog.js]: how about using 
```
hasUnfetchedSections = offset + limit < total;
```
?

Comment (VincentCATILLON) [reply src/containers/catalog.js]: it's not as simple as you mentions, you can have some unfetched sections between sections you already fetched, for example:

[true, true, false, false, true, true, true]

**Pull request merged**